src/core/data-abbr.js

```diff
@@ -0,0 +1,43 @@
+// @ts-check
+// Module core/data-abbr
+// - Finds all elements with data-abbr attribute and processes them.
+import { showInlineWarning } from "./utils";
+export const name = "core/dfn-abbr";
+
+export function run() {
+  document.querySelectorAll("[data-abbr]").forEach(elem => {
+    const { localName } = elem;
+    switch (localName) {
+      case "dfn":
+        // @ts-ignore
+        processDfnElement(elem);
+        break;
+      default: {
+        const msg =
+          `[\`data-dfn\`](https://github.com/w3c/respec/wiki/data-abbr)` +
+          ` attribute not supported on \`${localName}\` elements.`;
+        showInlineWarning(elem, msg, "Error: unsupported.");
+      }
+    }
+  });
+}
+/**
+ * @param {HTMLElement} dfn
+ */
+function processDfnElement(dfn) {
+  if (dfn.dataset.abbr === "") {
+    // Generates abbreviation from textContent
+    // e.g., "Permanent Account Number" -> "PAN"
+    dfn.dataset.abbr = dfn.textContent
```

Comment:
Do we need this assignment to `dfn.dataset.abbr`? Maybe we can do
``` js
const abbr = dfn.dataset.abbr || createAbbrFromText(fullForm);
```

Reply:
good suggestion